### ISSUE TITLE
Custom loss for TFModelV2

### ISSUE THREAD

**Issue by guzmo** [ray-project/ray#35905]
### What happened + What you expected to happen

1. I've implemented a custom TFModelV2 and tried to add a custom_loss() function, but it never gets triggered for TFModelV2. 
2. I expected that function to be called.
3. I have eager activated and when I print in different policies it seems the EagerTFPolicyV2 gets activated. Also, in the code I can also see that custom_loss is only triggered in TFPolicy but not in EagerTFPolicyV2. Am I missing something obvious here?

### Versions / Dependencies

ray 2.4.0
tensorflow 2.12.0

### Reproduction script

TFModelV2 with eager mode activated and a custom_loss function.

### Issue Severity

High: It blocks me from completing my task.

**Comment by RocketRider** [comment]
Did you test it without eager mode?

**Comment by guzmo** [comment]
How do I even disable eager mode when running with TF2? Tried different things like:
- tf.compat.v1.disable_eager_execution ()
- specifically say eager_traicing=False 
But ray still says.

` Executing eagerly (framework='tf2'), with eager_tracing=False. For production workloads, make sure to set eager_tracing=True  in order to match the speed of tf-static-graph (framework='tf'). For debugging purposes, `eager_tracing=False` is the best choice.`

**Comment by RocketRider** [comment]
You can set framework='tf' to force compat mode without eager_mode. I am still using this old way, as it seems the most stable one. In TF2 you can just change the eager_tracing=True/False setting.

**Comment by guzmo** [comment]
Yeah, with framework=tf custom_loss and update_ops() gets triggered :O so we need to stay with that?

**Comment by RocketRider** [comment]
Yes for now I would stay with "TF". For TF2 the ray team currently refactors the whole interface to use directly Keras models instead of own Model Class. You can test this new interface when you set the config: "_enable_learner_api": True and "_enable_rl_module_api": True.
But not sure how to use custom_loss with this API. Maybe you can look into that.

**Comment by RocketRider** [comment]
I guess you should not set the optimizer your self. This is setup in the policy itself. So no need for this. Then it should work....

**Comment by guzmo** [comment]
But I cant see that it compiles the model anywhere otherwise. And also that's a different loss then the custom_loss is it not?

**Comment by Rohan138** [comment]
Same as https://github.com/ray-project/ray/issues/35833#issuecomment-1599334364:

Either turn off `_enable_rl_module_api`, or migrate your custom model code to the new RLModule API.
